### ISSUE TITLE
yq throws TyperError datetime.datetime on ISOFormat strings in yaml file

### ISSUE THREAD

**Issue by jamesonjlee** [kislyuk/yq#5]
the default yaml loader will parse ISOFormat strings and create datetime objects which doesn't jsonify well.

---

Example:
```
$ yq --version
usage: yq [-h] [--yaml-output] [--width WIDTH] jq_filter [file [file ...]]
yq: error: too few arguments
$ pip freeze | grep yq
yq==2.1.2
$ yq . sec.yaml
yq: Error while running jq: TypeError: datetime.datetime(2016, 12, 20, 22, 7, 36) is not JSON serializable.
parse error: Unfinished JSON term
$ cat sec.yaml
data:
    thing: 1
    thing2: "stuff"
    datething: 2016-12-20T22:07:36Z
    datethingstr: "2016-12-20T22:07:36Z"
```

this needs to override the default encoder class.
```
# isoformat
class DateTimeEncoder(json.JSONEncoder):
    def default(self, o):
        if isinstance(o, datetime):
            return o.isoformat()
        return json.json.JSONEncoder.default(self, o)

# usage
json.dumps(..., cls=DateTimeEncoder)
```

**Comment by kislyuk** [comment]
Thanks! I can work on this tonight and cut a release, but if you wish to open a PR sooner, i can merge today.

**Comment by kislyuk** [comment]
Released in v2.2.0

**Comment by jamesonjlee** [comment]
looks fixed, thanks 👍
